src/goes/goes_converter.py

```diff
@@ -117,152 +143,266 @@ def _get_nadir_attribute_template(self):
+        lon_0 = goes_imager_projection.getncattr('longitude_of_projection_origin')
+        dataset_latlon = Dataset(self._latlon_file_path, 'r')
+        latitude = ma.getdata(dataset_latlon['/MetaData/latitude'][:].real)
+        longitude = ma.getdata(dataset_latlon['/MetaData/longitude'][:].real)
+        start_date = self._start_date
+        t = start_date.hour + (start_date.minute / 60.0) + (start_date.second / 3600.0)
+        h = -1.0 * ((t - 12.0) / 12.0)
+        beta_0 = np.arccos(np.cos(latitude) * np.cos(longitude - lon_0))
+        solar_azimuth_angle_data_array = np.arcsin((np.sin(h - longitude)) / (np.sin(beta_0)))
+        solar_azimuth_angle_data_array = \
+            self._goes_util.filter_data_array_by_yaw_flip_flag(solar_azimuth_angle_data_array)
+        solar_azimuth_angle_data_array = solar_azimuth_angle_data_array * 180.0 / np.pi
+        dataset.close()
+        dataset_latlon.close()
+        solar_azimuth_angle_data_array = np.nan_to_num(solar_azimuth_angle_data_array, nan=-999)
+        output_dataset.createVariable('/MetaData/solar_azimuth_angle', 'f4', 'nlocs', fill_value=-999)
+        output_dataset['/MetaData/solar_azimuth_angle'][:] = solar_azimuth_angle_data_array
+        output_dataset['/MetaData/solar_azimuth_angle'].setncattr('units', 'degrees')
+
+    def _create_metadata_elevation_angle_variable(self, output_dataset):
+        """
+        Creates the /MetaData/elevation_angle variable in an output netCDF4 dataset.
+        output_dataset - A netCDF4 Dataset object
+        """
+        elevation_angle_data_array = self._latlon_dataset['MetaData'].variables['elevation_angle'][:].real
+        output_dataset.createVariable('/MetaData/elevation_angle', 'f4', 'nlocs', fill_value=-999)
+        output_dataset['/MetaData/elevation_angle'][:] = elevation_angle_data_array
+
+    def _create_nlocs_dimension(self, output_dataset):
+        """
+        Creates the nlocs dimension in an output netCDF4 dataset.
+        output_dataset - A netCDF4 Dataset object
         """
         nlocs = self._latlon_dataset.dimensions['nlocs'].size
-        self._output_dataset_rf.createDimension('nlocs', nlocs)
-        self._output_dataset_rf.createVariable('nlocs', 'i4', 'nlocs')
-        self._output_dataset_rf.variables['nlocs'].setncattr('suggested_chunk_dim', nlocs)
-        self._output_dataset_rf.variables['nlocs'][:] = np.arange(1, nlocs + 1, 1, dtype='int32')
-        self._output_dataset_bt.createDimension('nlocs', nlocs)
-        self._output_dataset_bt.createVariable('nlocs', 'i4', 'nlocs')
-        self._output_dataset_bt.variables['nlocs'].setncattr('suggested_chunk_dim', nlocs)
-        self._output_dataset_bt.variables['nlocs'][:] = np.arange(1, nlocs + 1, 1, dtype='int32')
+        output_dataset.createDimension('nlocs', nlocs)
+        output_dataset.createVariable('nlocs', 'i4', 'nlocs')
+        output_dataset.variables['nlocs'].setncattr('suggested_chunk_dim', nlocs)
```

Comment:
I know the chunk_dim is used for the compression algorithm, but shouldn't it be set lower than the number of total locations?  Isn't it more optimal to chunk the vector is smaller units that compress smaller by using semi-local min/max/range info?

Reply:
This is a placeholder until an optimized chuck dimension has been selected. This will be addressed in a future PR.